balancer-js/src/modules/pools/pool-types/concerns/types.ts

```diff
@@ -13,3 +15,69 @@ export interface SpotPriceConcern {
     pool: SubgraphPoolBase
   ) => string;
 }
+
+export interface JoinConcern {
+  buildExactTokensInJoinPool: ({
+    joiner,
+    pool,
+    tokensIn,
+    amountsIn,
+    slippage,
+  }: ExactTokensInJoinPoolParameters) => Promise<JoinPoolAttributes>;
+}
+
+export interface ExitConcern {
+  buildExitExactBPTInForTokensOut: ({
+    exiter,
+    pool,
+    bptIn,
+    slippage,
+  }: ExitExactBPTInForTokensOutParameters) => Promise<ExitPoolAttributes>;
+}
+
+export interface JoinPool {
+  poolId: string;
+  sender: string;
+  recipient: string;
+  joinPoolRequest: JoinPoolRequest;
+}
+
+export interface JoinPoolAttributes {
+  to: string;
+  functionName: string;
+  attributes: JoinPool;
+  data: string;
+  value?: BigNumber;
+  minAmountsOut: string;
```

Comment:
Rename minAmountsOut to something like minBptOut.

Reply:
Agree - done ✅